osu.Game.Tests/Visual/Gameplay/TestScenePause.cs

```diff
@@ -82,35 +94,43 @@ public void TestPauseWithLargeOffset()
                 offset.Value = -5000;
             });
 
+            hookForwardPlaybackCheck();
+
+            AddStep("move cursor outside", () => InputManager.MoveMouseTo(Player.ScreenSpaceDrawQuad.TopLeft - new Vector2(10)));
+
+            pauseAndConfirm();
+
+            resumeAndConfirm();
+
+            checkForwardPlayback();
+
+            AddStep("reset offset", () => LocalConfig.SetValue(OsuSetting.AudioOffset, 0.0));
+        }
+
+        private void checkForwardPlayback() => AddAssert("time didn't go too far backwards", () => gameplayClockAlwaysGoingForward);
+
+        private void hookForwardPlaybackCheck()
+        {
             AddStep("add time forward check hook", () =>
             {
-                lastStopTime = double.MinValue;
-                alwaysGoingForward = true;
+                lastForwardCheckTime = double.MinValue;
+                gameplayClockAlwaysGoingForward = true;
 
                 Player.OnUpdate += _ =>
                 {
-                    var masterClock = (MasterGameplayClockContainer)Player.GameplayClockContainer;
+                    var frameStableClock = Player.ChildrenOfType<FrameStabilityContainer>().Single().Clock;
```

Comment:
This uses frame stable time for the tests (even the existing usage). This is a bit different to the previous guarantee, but the previous one *was never hooked up* – `lastStopTime` was never updated – so I'm not really caring at this point.

Reply:
> `lastStopTime` was never updated

wooooooooooooooooooow. incredible